### PR TITLE
Create subscription page (copy of #6647 but with a clean commit history)

### LINKED ISSUE #6582: T - Create landing page (FAQ) for Subscriptions
## 💬 Description
Create a landing page for Subscriptions in React, so we can reuse components. The page will live at /payments. 

## ✅ Acceptance Criteria
- [ ] User can navigate to /payments page
- [ ] User can read FAQ about why we are doing this and what to expect 
- [ ] User viewing on a smaller screen does not have a terrible experience

_Note: The "Help grow MDN" callout in the header is **not** expected to be completed during this sprint_
 
## 👷‍♀ Tasks
- [x] Add a new route and view in Django for /payments
- [x] Create React component, borrowing content from existing [Thank You page](https://wiki.developer.mozilla.org/en-US/payments/recurring)
- [x] Update copy to match what's in [latest copy edits](https://docs.google.com/document/d/1TH-P-hcZY1UzBtXDGGZnF3jTCy7b-cgC3TUo4Vwxfdc/edit?usp=sharing )

## 🎨 Design
https://app.abstract.com/projects/9decc480-70fc-11e9-9453-4f3fbffea7ce/branches/master/commits/5eb3004bb0e0aeb362aae44b3da2fa7d15d893cb/files/96E76C7B-2AF8-4473-A39E-E81890F91955/layers/0A8C3971-7D07-40BF-96B5-3B467D450CC4?mode=build&selected=734966551-1AF136E1-5788-4D35-A3CF-0E452D009A7B

~~https://mozilla.invisionapp.com/share/TDVED747SB9#/screens/399165366~~

## 🔦 Context
Please see Schalk's comment https://github.com/mdn/kuma/issues/6581#issuecomment-591376908

### PR DESCRIPTION
**Pull request by mindy**
**Summary:** 
Convert existing FAQ page to the Subscription landing page in React. 

**To do:** 
- [ ] Translations are not working (see screenshots below, where locale is `FR` and header is translated, but not the main content.

**URLs** 
**/payments** is changing from wiki to non-wiki
**/contribute** currently redirects to /payments, is changing from wiki to non-wiki 
**/payments/recurring** is still on the wiki domain, same old Jinja page for now. Header says "Thank you for supporting MDN."

**To test:** 
Go to /payments (http://localhost.org:8000/en-US/payments): 
Screen Shot 2020-03-05 at 5 29 40 PM

Make sure former /payments/recurring page still works by going to /payments/recurring:
Screen Shot 2020-03-05 at 5 29 47 PM
 
Closes #6582

### CODE REVIEW

**Comment by schalkneethling** [comment]
@mindy Thank you for the PR!

There is one Python test that is currently failing:

```
=================================== FAILURES ===================================
___________________________ test_redirect[payments] ____________________________
Traceback (most recent call last):
  File "/app/kuma/payments/tests/test_views.py", line 181, in test_redirect
    assert_redirect_to_wiki(response, url)
  File "/app/kuma/core/tests/__init__.py", line 14, in assert_redirect_to_wiki
    assert response.status_code == 301
AssertionError
----------------------------- Captured stdout call -----------------------------
<class 'requests.exceptions.ConnectionError'> error contacting SSR server.
Falling back to client side rendering.
================== 1 failed, 1967 passed in 193.13s (0:03:13) ==================
make: *** [Makefile:36: coveragetest] Error 1
```

You can run these by following the steps here:
https://kuma.readthedocs.io/en/latest/tests.html#running-the-test-suite

NOTE: The first step is to do `make bash` and then run the rest of the commands from the resulting bash. To avoid having to run all of the tests, you can do the following for the above error:

```
py.test kuma/payments/
```

**Comment by mindy** [comment]
@schalkneethling Thank you so much for your help! I fixed the test by removing "payments" from that array, since /payments will be not on the wiki so I don't think the test applies to that page any longer.

File: kuma/urls.py [hunk 1]
```diff
@@ -105,6 +102,9 @@
     ),
 )
 urlpatterns += i18n_patterns(re_path(r"^payments/", include(payments_lang_urlpatterns)))
+urlpatterns += i18n_patterns(
+    re_path(r"payments$", payment_views.index, name="payments")
+)
```
Comment (peterbe) [inline kuma/urls.py]: I think we need to think deeper about this. It's nice that we're being consistent and not using a trailing slash if other pages don't use that. However, `/payments/` is actually a namespace. Look at the line above (line 104). 
So what you have name is a clash. 
Basically, I think the `payments_view.index` needs to be moved into `payments/urls.py`.

**Comment by peterbe** [comment]
I don't like that the document title is "Contribute ..." when the page isn't about contributing. It's about frequently asked questions around *subscriptions*.
Screen Shot 2020-03-10 at 11 36 51 AM

**Comment by peterbe** [comment]
I think this PR needs to also clean up the old payments FAQ. 
If this PR merges we'll have the same information written twice. Once in a jinja template and once in a React .jsx file.

**Comment by peterbe** [comment]
There's something strange going on and it's bizarre that nothing visually is happening. 
This is what I get when I visit http://localhost.org:8000/fr/payments?a=d
Screen Shot 2020-03-10 at 11 44 12 AM

**Comment by peterbe** [comment]
Now I know what's going on. That page is actually never server-rendered. This is a nasty trick that Kuma's React does and I've never understood or liked it. 
If you curl it this is what happens:
```
▶ curl -I http://localhost.org:8000/fr/payments
HTTP/1.1 404 Not Found
Date: Tue, 10 Mar 2020 15:47:51 GMT
Server: WSGIServer/0.2 CPython/3.8.0
Content-Type: text/html
X-Frame-Options: DENY
Content-Length: 1709
Content-Language: fr
Vary: Cookie
X-Content-Type-Options: nosniff
X-XSS-Protection: 1; mode=block
```

As if the page doesn't exist. Fair. It means we have a problem but you don't see a loud "Sorry! Page Does Not Exist". Instead, what it does it that it falls back on doing a client-side render. 
Let's not do that. SSR render these pages. (which very much might explain the confusion about never-localized strings)

Also, how come no test noticed this? I think that's the first thing to look into. Opening `client.get('/en-US/payments')` in a test should be 200 OK and the content of it should have stuff about the FAQ.

**Comment by mindy** [comment]
> I don't like that the document title is "Contribute ..." when the page isn't about contributing. It's about frequently asked questions around _subscriptions_.

Changed it to "Become a monthly supporter". Also eventually this page will have a CTA in the header to contribute: 
contribute

**Comment by mindy** [comment]
> I think this PR needs to also clean up the old payments FAQ.
> If this PR merges we'll have the same information written twice. Once in a jinja template and once in a React .jsx file.

I see that, but I think it's fine to leave it for now-- in the next sprint I'm going to convert the existing Jinja page to something else: https://github.com/mdn/kuma/issues/6581

**Comment by mindy** [comment]
@peterbe I need some guidance on how to do this-- do you have an example of an SSR page I can look at for an example?

**Comment by peterbe** [comment]
> > I think this PR needs to also clean up the old payments FAQ.
> > If this PR merges we'll have the same information written twice. Once in a jinja template and once in a React .jsx file.
> 
> I see that, but I think it's fine to leave it for now-- in the next sprint I'm going to convert the existing Jinja page to something else: #6581

In that case, I'd rather you put a note in the old stuff to say that it's going to be deleted.

**Comment by peterbe** [comment]
> @peterbe I need some guidance on how to do this-- do you have an example of an SSR page I can look at for an example?

The site-search thing. Do what it does. But you don't have a `result` so it might need some work. I'm going to attempt to do it too, in parallel, and we can compare results.

**Comment by mindy** [comment]
> > > I think this PR needs to also clean up the old payments FAQ.
> > > If this PR merges we'll have the same information written twice. Once in a jinja template and once in a React .jsx file.
> > 
> > 
> > I see that, but I think it's fine to leave it for now-- in the next sprint I'm going to convert the existing Jinja page to something else: #6581
> 
> In that case, I'd rather you put a note in the old stuff to say that it's going to be deleted.

Added a comment at the top of payments.html

**Comment by peterbe** [comment]
Another thing, before I forget, I don't like that the "payment terms" page is on the Wiki only. I can appreciate that you have to go to the Wiki to do your Stripe thing, for now, but I think the terms and the faq ought to be on the same site.

File: jinja2/react_base.html [hunk 2]
```diff
@@ -104,33 +104,35 @@
...
+            <div id="license" class="contentinfo">
+              {% trans copyright_url='/en-US/docs/MDN/About#Copyrights_and_licenses', thisyear=thisyear() %}
+                <p>&copy; 2005-{{ thisyear }} Mozilla and individual contributors.</p>
+                <p>Content is available under <a href="{{ copyright_url }}">these licenses</a>.</p>
+              {% endtrans %}
+            </div>
+          {% endblock %}
+      </div>
+    </footer>
+  {% endblock %}
```
Comment (peterbe) [inline jinja2/react_base.html]: I still cannot understand why this change is necessary? 
As far as I can tell, the block of code was put in a `{% block footer %} ... {% endblock %}` But is there any reason to do it?

File: kuma/javascript/src/payments/index.jsx [hunk 3]
```diff
@@ -0,0 +1,367 @@
...
+    }
+
+    getComponent() {
+        return PaymentsLandingPage;
+    }
+
+    match(url: string): ?PaymentsRouteParams {
+        let path = new URL(url, BASEURL).pathname;
+
+        if (path !== `/${this.locale}/payments`) {
```
Comment (peterbe) [inline kuma/javascript/src/payments/index.jsx]: We just need to be very careful that this works when we figure out the correct URL trailing slash thing in `urls.py`.

File: kuma/javascript/src/payments/subheader.jsx [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+import * as React from 'react';
+
+import { gettext } from '../l10n.js';
+
+type Props = {
+    title: string,
+    subtitle?: string,
+    description?: string,
+    children?: React.Node,
+    columnWidth?: string
```
Comment (peterbe) [inline kuma/javascript/src/payments/subheader.jsx]: Is it really optional?

File: kuma/payments/jinja2/payments/index.html [hunk 5]
```diff
@@ -1,9 +1,14 @@
 {% extends "react_base.html" %}
 
-{% block title %}{{ _('Become a monthly supporter') }}{% endblock %}
+{% block title %}{{ page_title(_('Become a monthly supporter')) }}{% endblock %}
 
 {% block document_head %}
-    {{ render_react("SPA", request.LANGUAGE_CODE, request.get_full_path(), None)|safe }}
+    <!--
```
Comment (peterbe) [inline kuma/payments/jinja2/payments/index.html]: This needs to be `{#` and `#}` otherwise it becomes a HTML comment.

File: kuma/payments/jinja2/payments/index.html [hunk 6]
```diff
@@ -1,9 +1,14 @@
...
+        This page doesn't need to fetch data, so we pass `True`
+        as an argument to indicate that we have the data we need, which
+        allows the component to load properly.
```
Comment (peterbe) [inline kuma/payments/jinja2/payments/index.html]:
```suggestion
        allows the component render in the server. 
```

File: kuma/urls.py (hunk 1 above)
Comment (mindy) [reply kuma/urls.py]: Will work on this in the next few minutes 👍

File: kuma/javascript/src/payments/subheader.jsx (hunk 4 above)
Comment (mindy) [reply kuma/javascript/src/payments/subheader.jsx]: Yeah, because we set a default value for `columnWidth` ("all") below

File: jinja2/react_base.html (hunk 2 above)
Comment (mindy) [reply jinja2/react_base.html]: If we don't have this, it renders two footers. So this allows us to override this footer in `payments/index.html`:

two-footers

Comment (mindy) [reply jinja2/react_base.html]: I could remove the React Footer and leave the one here, but when do we know which one to use? The React Footer is only used in documents right now.

File: kuma/payments/jinja2/payments/index.html (hunk 5 above)
Comment (mindy) [reply kuma/payments/jinja2/payments/index.html]: Fixed. Thank you!

File: kuma/urls.py (hunk 1 above)
Comment (mindy) [reply kuma/urls.py]: This is now fixed-- the index page is now being handled in `payments/urls.py`. Thanks for the suggestion!

**Comment by schalkneethling** [comment]
Hey @mindy,

Not to throw a spanner in the works but what was the reason for not using the React-based footer here?

The build is failing because of this:

```
kuma/urls.py:17:1: F401 'kuma.payments.views as payment_views' imported but unused
```

You can safely remove that `import`.

Also as a side note. If any/all of the "instructions" I provided was incorrect or contributed to these problems, let's update it so we have a nice how-to for when we need to do this next.

Thanks so much @mindy! 🍵

**Comment by mindy** [comment]
Thanks @schalkneethling ! I removed the React footer because I wasn't sure which one to use (https://github.com/mdn/kuma/pull/6660#discussion_r390573927) and the React footer left a gap at the bottom of the page (which is fixed now): 

Screen Shot 2020-03-11 at 11 24 35 AM

I added the React footer back in, and moved the `<div class="center clear"></div>` from `react_base.html` to the specific components that use `block content` (it was only the homepage). 

Great idea-- I'll add a page to the Kuma wiki with your instructions and some edits.

File: jinja2/react_base.html (hunk 2 above)
Comment (peterbe) [reply jinja2/react_base.html]: I give up on all this footer stuff. I think you an @schalkneethling should push back on the influx of feature additions and convince management that you have to refactor this mess up. 
As mentioned in Slack, we have **4** `<html ...` tags, when it should be 2 (one for the Wiki and one for the read-only site). 
```
▶ rg '<html ' | rg -v jquery | rg -v ckeditor | rg -v tests | rg -v '\.ltxt'
jinja2/base.html:<html lang="{% block lang %}{{ LANG }}{% endblock %}" dir="{{ DIR }}" class="no-js">
jinja2/react_base.html:<html lang="{% block lang %}{{ LANG }}{% endblock %}" dir="{{ DIR }}" class="no-js">
kuma/wiki/jinja2/wiki/react_document.html:<html lang="{{ document_data['hrefLang'] }}" dir="{{ DIR }}">
kuma/users/jinja2/socialaccount/base_signup.html:<html lang="{% block lang %}{{ LANG }}{% endblock %}" dir="{{ DIR }}" class="no-js">
```
And **4** `<footer` tags when there should be 2. 
```
▶ rg '<footer' | rg -v jquery | rg -v ckeditor | rg -v tests | rg -v '\.ltxt' | rg -v l10n.js
jinja2/base.html:    <footer id="nav-footer" class="nav-footer">
jinja2/react_base.html:    <footer id="nav-footer" class="nav-footer">
kuma/javascript/src/footer.jsx:        <footer id="nav-footer" className="nav-footer">
kuma/users/jinja2/socialaccount/base_signup.html:    <footer id="nav-footer" class="nav-footer">
```

File: kuma/javascript/src/payments/subheader.jsx (hunk 4 above)
Comment (peterbe) [reply kuma/javascript/src/payments/subheader.jsx]: bah! I meant the line above. `children` Sorry. Really not sure it matters at all.

File: kuma/payments/jinja2/payments/payments.html [hunk 7]
```diff
@@ -1,3 +1,5 @@
+<!-- This page will be replaced in the next sprint by https://github.com/mdn/kuma/issues/6581 -->
```
Comment (peterbe) [inline kuma/payments/jinja2/payments/payments.html]:
```suggestion
{# This page will be replaced in the next sprint by https://github.com/mdn/kuma/issues/6581 #}
```

File: kuma/javascript/src/payments/index.jsx [hunk 8]
```diff
@@ -0,0 +1,363 @@
...
+        if (regex.test(path)) {
+            return {
+                locale: this.locale
+            };
+        }
+        return null;
+    }
+
+    fetch() {
+        return Promise.resolve(null);
```
Comment (peterbe) [inline kuma/javascript/src/payments/index.jsx]: I think this should never run because we never need to do client-side fetch. So you can replace this with 
```suggestion
        throw new Error('Payments should never need to post-fetch more data');
```

File: kuma/javascript/src/payments/index.jsx [hunk 9]
```diff
@@ -0,0 +1,363 @@
...
+    getComponent() {
+        return PaymentsLandingPage;
+    }
+
+    match(url: string): ?PaymentsRouteParams {
+        const path = new URL(url, BASEURL).pathname;
+        const paymentsPath = `/${this.locale}/payments`;
+        const regex = new RegExp(paymentsPath, 'g');
+
+        if (regex.test(path)) {
```
Comment (peterbe) [inline kuma/javascript/src/payments/index.jsx]: Pardon my ignorance but why is this necessary? Why can't it be `if (path.startsWith(`/${this.locale}/payments`) {`?

**Comment by peterbe** [comment]
It might seem silly but it's hugely beneficial for possible future refactoring if you at least render the view. 
https://github.com/mindy/kuma/pull/1

File: jinja2/react_base.html (hunk 2 above)
Comment (schalkneethling) [reply jinja2/react_base.html]: Ah ok, I am on the same page here now. The change was indeed simple to wrap the `footer` HTML inside a `footer` Jinja block. This is required because `payments/index.html` extends `react_base.html`. If the `footer` is not wrapped in a `block` there is no way for us to override the `block` inside `payments/index.html`

The result would be two footers. One from React and one from the extended `react_base.html`. The reason `search/react.html` did not run into this scenario first is that it does not use the React footer but the one inherited from `react_base.html`

This means that as long as we have pages that need to extend `react_base.html` the following will always be the template used by said page:

```
{% extends "react_base.html" %}

{% block title %}{{ page_title(_('Become a monthly supporter')) }}{% endblock %}

{% block document_head %}
   {#
        This page doesn't need to fetch data, so we pass `True`
        as an argument to indicate that we have the data we need, which
        allows the component to render in the server.
    #}
    {{ render_react("SPA", request.LANGUAGE_CODE, request.get_full_path(), True)|safe }}
{% endblock %}

{% block content %}{% endblock %}

{% block footer %}{% endblock %}
```

SPA is probably not the best naming scheme anymore inside Kuma anyway. RPC(React Page Component) might be more appropriate :)

Comment (schalkneethling) [reply jinja2/react_base.html]: With that said, if all pages that extend `react_base.html` do not need the footer any longer, we can remove the footer from `react_base.html` and in so doing, the need for the Jinja `footer block`.

Currently, though, three pages extend it:

1. `landing/react_homepage.html`
2. `payments/index.html`
3. `search/react.html`

Numbers 2 and 3 do not need the footer but, until the homepage is converted to React, it will still need it.

File: kuma/static/styles/components/payments/_page.scss [hunk 10]
```diff
@@ -117,6 +119,10 @@
                 padding-left: 1em;
```
Comment (schalkneethling) [inline kuma/static/styles/components/payments/_page.scss]: I would suggest that we add `margin-top: 20px;` here to create a bit of space between the paragraph and the list.

File: kuma/javascript/src/payments/subheader.jsx (hunk 4 above)
Comment (mindy) [reply kuma/javascript/src/payments/subheader.jsx]: I see-- when I remove `children`, it complains that that it wasn't in the Props declaration, so we need it here.

Comment (mindy) [reply kuma/javascript/src/payments/subheader.jsx]: It does-- I got a complaint when I removed the `?` saying that children was required in `<SubHeader />` but not provided

File: kuma/static/styles/components/payments/_page.scss (hunk 10 above)
Comment (mindy) [reply kuma/static/styles/components/payments/_page.scss]: Changed the `p:last-of-type` rule to `p:last-child` rule instead, so we still have the margin between list and paragraph. Thank you for pointing this out!

File: kuma/javascript/src/payments/index.jsx (hunk 9 above)
Comment (mindy) [reply kuma/javascript/src/payments/index.jsx]: Good point-- changed to `startsWith` instead. Thanks Peter!

**Comment by mindy** [comment]
Ready for another 👀 when you get a chance, @schalkneethling

File: jinja2/react_base.html (hunk 2 above)
Comment (mindy) [reply jinja2/react_base.html]: Opened up https://github.com/mdn/kuma/issues/6678 so we don't loose track of this.
